core/types/crosslink.go

```diff
@@ -0,0 +1,44 @@
+package types
+
+import (
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// CrossLink is only used on beacon chain to store the hash links from other shards
+type CrossLink struct {
+	shardID  uint32
+	blockNum uint64
+	hash     common.Hash
```

Comment:
TODO: Add state root and cxreceipt root.

Reply:
since you are going to use header instead, I will just ignore this comment.